tests/sanity/sanity_test.go

```diff
@@ -0,0 +1,121 @@
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/golang/mock/gomock"
+	csisanity "github.com/kubernetes-csi/csi-test/v5/pkg/sanity"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/cloud/metadata"
+	d "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	instanceID       = "i-1234567890abcdef0"
+	region           = "us-west-2"
+	availabilityZone = "us-west-2a"
+)
+
+var (
+	fakeMetadata = &metadata.Metadata{
+		InstanceID: instanceID,
+		Region:     region,
+	}
+	outpostArn = &arn.ARN{
+		Partition: "aws",
+		Service:   "outposts",
+		Region:    "us-west-2",
+		AccountID: "123456789012",
+		Resource:  "op-1234567890abcdef0",
+	}
+)
+
+func TestSanity(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Test panicked: %v", r)
+		}
+	}()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tmpDir, err := os.MkdirTemp("", "csi-sanity-")
+	if err != nil {
+		t.Fatalf("Failed to create sanity temp working dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	defer func() {
+		if err = os.RemoveAll(tmpDir); err != nil {
+			t.Fatalf("Failed to clean up sanity temp working dir %s: %v", tmpDir, err.Error())
+		}
+	}()
+
+	endpoint := fmt.Sprintf("unix:%s/csi.sock", tmpDir)
+	mountPath := path.Join(tmpDir, "mount")
+	stagePath := path.Join(tmpDir, "stage")
+
+	fakeMounter, fakeCloud, fakeMetadataService, fakeKubernetesClient := createMockObjects(mockCtrl, mountPath)
+
+	driverOptions := &d.Options{
+		Mode:                              d.AllMode,
+		ModifyVolumeRequestHandlerTimeout: 60,
+		Endpoint:                          endpoint,
+	}
+
+	drv, err := d.NewDriver(fakeCloud, driverOptions, fakeMounter, fakeMetadataService, fakeKubernetesClient)
+	if err != nil {
+		t.Fatalf("Failed to create fake driver: %v", err.Error())
+	}
+	go func() {
+		if err := drv.Run(); err != nil {
+			panic(fmt.Sprintf("%v", err))
+		}
+	}()
+
+	config := csisanity.TestConfig{
+		TargetPath:                  mountPath,
+		StagingPath:                 stagePath,
+		Address:                     endpoint,
+		DialOptions:                 []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())},
+		IDGen:                       csisanity.DefaultIDGenerator{},
+		TestVolumeSize:              10 * util.GiB,
+		TestVolumeAccessType:        "mount",
+		TestVolumeMutableParameters: map[string]string{"iops": "3014", "throughput": "153"},
+		TestVolumeParameters:        map[string]string{"type": "gp3", "iops": "3000"},
+	}
+	csisanity.Test(t, config)
+}
+
+func createMockObjects(mockCtrl *gomock.Controller, mountPath string) (*fakeMounter, *fakeCloud, *fakeMetadataService, *d.MockKubernetesClient) {
+	fakeMounter := NewFakeMounter()
+	fakeCloud := NewFakeCloud(*fakeMetadata, mountPath)
+	fakeKubernetesClient := d.NewMockKubernetesClient(mockCtrl)
```

Comment:
If we're removing mock objects, let's remove all of them. 😄 

(I don't want to block the PR over this though)

Reply:
Removed it entirely as it is not actually needed.